### PR TITLE
[IR-3322] studio: update ArrayInputGroup to improve usability and accept dropped files

### PR DESCRIPTION
**Pull request by dtlehrer**
## Summary
[IR-3322] now able to drag & drop files into ArrayInputGroup

## References
closes #_insert number here_

## QA Steps

[IR-3322]: https://tsu.atlassian.net/browse/IR-3322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dinomut1 — DISMISSED** [review]
[Screencast from 07-23-2024 03:27:34 PM.webm](https://github.com/user-attachments/assets/8c8a91e8-7d66-4de3-b69f-5f39910af1e2)
Can't drag and drop onto individual array items, instead it just adds it to the list. 
There should be an additional drop area in the URL field of each individual list item that lets you change the url for that item

**Comment by dtlehrer** [comment]
> [Screencast from 07-23-2024 03:27:34 PM.webm](https://github.com/user-attachments/assets/8c8a91e8-7d66-4de3-b69f-5f39910af1e2) Can't drag and drop onto individual array items, instead it just adds it to the list. There should be an additional drop area in the URL field of each individual list item that lets you change the url for that item

updated to handle this case as well:
https://github.com/user-attachments/assets/29cf7a40-cb70-4065-aee9-0647b333b854

File: packages/ui/src/components/editor/input/Array/index.tsx [hunk 1]
```diff
@@ -36,17 +38,44 @@ export interface ArrayInputProps {
...
+    },
+    collect: (monitor) => ({
+      isDroppable: monitor.canDrop() && monitor.isOver()
+    })
+  }))
+
+  return (
+    <div className="flex flex-col px-3">
+      {inputLabel && <Label className="mb-1 text-[#A0A1A2]">{inputLabel + ' ' + (index + 1)}</Label>}
+      <div ref={dropRef} className={`mb-2 flex items-center ${isDroppable ? 'outline outline-2 outline-white' : ''}`}>
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/input/Array/index.tsx]:
```suggestion
      <div ref={dropRef} className={twMerge('mb-2 flex items-center', isDroppable && 'outline outline-2 outline-white')}>
```

we can utilise twMerge in this cases - it simplifies using normal string

File: packages/ui/src/components/editor/input/Array/index.tsx [hunk 2]
```diff
@@ -36,17 +38,44 @@ export interface ArrayInputProps {
   values: string[]
   onChange: (values: string[]) => void
   inputLabel?: string
+  dropTypes?: string[]
 }
 
-// TODO: file and drag and drop functionality
+const DiscardableInput = ({ value, index, inputLabel, onChange, onRemove, dropTypes }) => {
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/input/Array/index.tsx]:
```suggestion
const DiscardableInput = ({
  value,
  index,
  inputLabel,
  onChange,
  onRemove,
  dropTypes
}: {
  value: string
  index: number
  onChange: (val: string, idx: number) => void
  onRemove: (idx: number) => void
} & Pick<ArrayInputProps, 'inputLabel' | 'dropTypes'>) => {
```

**Review by aditya-mitra — CHANGES_REQUESTED** [review]
seeing an anamoly when updating the first item

https://github.com/user-attachments/assets/5efa38f9-5138-45be-9a8b-75d8d2295384

**Review by aditya-mitra — APPROVED** [review]
awesome bug solving

**Pull request merged**